zerver/lib/thumbnail.py

```diff
@@ -333,11 +333,14 @@ class MarkdownImageMetadata:
 
 
 def get_user_upload_previews(
-    realm_id: int, content: str
+    realm_id: int, content: str, lock: bool = False
 ) -> dict[str, MarkdownImageMetadata | None]:
     matches = re.findall(r"/user_uploads/(\d+/[/\w.-]+)", content)
```

Comment:
We might want an `if len(matches) == 0: return None` escape hatch here, to avoid doing pointless queries in the case that there are no matches. At least, I think Django will end up issuing SQL for the queries below?

Oh, I guess maybe that's impossible, in that `do_claim_attachments` actually claimed some.

Reply:
Yeah, I intentionally tucked it inside there.